### ISSUE TITLE
Failed to run on MacOS for a shader compilation 

### ISSUE THREAD

**Issue by n826vnl8** [Ruddle/oxidator#1]
I got below message on my MacOS. Please let me know if you need more information.

- Error Message:

```
[2020-02-21T05:23:57Z ERROR gfx_backend_metal::device] Error compiling the shader CompilationFailed("ImageQueryLod is only supported on MSL 2.2 and up.")
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Other', src/libcore/result.rs:1188:5
stack backtrace:
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: std::panicking::rust_panic_with_hook
   6: rust_begin_unwind
   7: core::panicking::panic_fmt
   8: core::result::unwrap_failed
   9: wgpu_device_create_render_pipeline
  10: wgpu::Device::create_render_pipeline
  11: oxidator::gpu_obj::heightmap_gpu::HeightmapGpu::create_pipeline
  12: oxidator::gpu_obj::heightmap_gpu::HeightmapGpu::new
  13: oxidator::client::App::new
  14: oxidator::main
  15: std::rt::lang_start::{{closure}}
  16: std::panicking::try::do_call
  17: __rust_maybe_catch_panic
  18: std::rt::lang_start_internal
  19: main

```

**Comment by kvark** [comment]
This is fixed upstream in https://crates.io/crates/gfx-backend-metal/0.4.3
Note that #3 also needs to be fixed in order for the game to run.

**Comment by Ruddle** [comment]
Thank you for the report, and the fix. I bumped wgpu-native to 4.3 and fixed #3 .
Feel free to open again if the same error occurs.
